highway_env/envs/common/abstract.py

```diff
@@ -383,6 +383,64 @@ def __deepcopy__(self, memo):
                 setattr(result, k, None)
         return result
 
+class MOAbstractEnv(AbstractEnv):
+    """
+    A multi-objective version of AbstractEnv. 
+    Environments should inherit from MOAbstractEnv and register new reward callbacks.
+    Inheriting classes must also define a utility function to produce a scalar reward from the reward vector.
+
+    Note: The vector of rewards is returned by _info() and the scalar utility is return by _reward()
+    """
+    def __init__(self, config: dict = None) -> None:
```

Comment:
Very true, for some reason I felt like the multi objective case would require more extensive changes so I broke it out into a new class but your way is much simpler. I still think there may be some benefit to storing a `self.rewards` value in the class so that `_reward()` and `_info()` don't call the same method twice, but with simple rewards it doesn't matter too much.